File: 0009-palindrome-number/0009-palindrome-number.go
import "strconv"

func isPalindrome(x int) bool {
    if x < 0 {
        return false
    }

    s := strconv.Itoa(x)
    left := 0
    right := len(s) - 1
    for left < right {
        if s[left] != s[right] {
            return false
        }
        left++
        right--
    }

    return true
}

File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0 || (x > 0 && x % 10 == 0)) {
            return false;
        }

        int reserved = 0;
        while (reserved < x) {
            reserved = reserved * 10 + x % 10;
            x /= 10 ;
        }

        return (x == reserved) || (x == reserved / 10);
    }
};